.github/workflows/dev_module_build-on-self-hosted-runner.yml

```
@@ -32,24 +32,31 @@ jobs:
         name: Check label
         run: echo "should_run=true"  >> $GITHUB_OUTPUT
 
-  show_dev_manifest:
-    needs: check_label
-    if: needs.check_label.outputs.should_run
-    runs-on: [self-hosted, ci-testing]
-    name: Show manifest
+  set_vars:
+    runs-on: ubuntu-latest
+    name: Set MODULES_MODULE_TAG
+    outputs:
+      modules_module_tag: ${{ steps.modules_module_tag.outputs.MODULES_MODULE_TAG }}
     steps:
-      - name: Set vars for PR
-        if: ${{ github.ref_name != 'main' }}
+      - name: Set vars
+        id: modules_module_tag
         run: |
-          MODULES_MODULE_TAG="$(echo ${{github.event.pull_request.number}})"
-          echo "MODULES_MODULE_TAG=$MODULES_MODULE_TAG" >> "$GITHUB_ENV"
+          if [[ "${{ github.ref_name }}" == 'pre-alpha' || "${{ github.ref_name }}" == 'main' ]]; then
+            MODULES_MODULE_TAG="${{ github.ref_name }}"
+          else
+            MODULES_MODULE_TAG="$(echo pr${{ github.event.pull_request.number }})"
+          fi
 
-      - name: Set vars for main
-        if: ${{ github.ref_name == 'main' }}
-        run: |
-          MODULES_MODULE_TAG="$(echo v0.0.0-${{ github.ref_name }})"
-          echo "MODULES_MODULE_TAG=$MODULES_MODULE_TAG" >> "$GITHUB_ENV"
+          echo "MODULES_MODULE_TAG=$MODULES_MODULE_TAG" >> "$GITHUB_OUTPUT"
 
+  show_dev_manifest:
+    name: Show manifest
+    if: needs.check_label.outputs.should_run
+    needs: set_vars
```

```suggestion
    if: needs.check_label.outputs.should_run
    needs:
    - check_label
    - set_vars
```

Add this to other steps.